### PR TITLE
feat(erc20): dynamic erc20 list

### PR DESCRIPTION
**Pull request by plondon**
## Phase 1
- [x] dynamic erc20 balances
- [x] find and remove Erc20CoinsEnum
- [x] find and remove Erc20ListEnum
- [x] finish // TODO: convertCoinToFiat when standard to standard
- [x] only require a precision, instead of a coin object/file
- [x] convertCoinToFiat, should accept an object, check TODO: convertCoinToFiat when standard to standard
- [x] pull erc20s from new coin config api, combine w/ wallet options
- [x] supported coins should include erc20s you have a balance for
- [x] remove coin availabilities, eventually endpoint will determine this
- [x] Wallet/MenuTop/Features
- [x] Remove erc20s from wallet options
- [x] make supported coins truly remote and add erc20s w/ balance to that
- [x] send coin dropdown should not include fiat currencies
- [x] fix NaN on request screen
- [x] https://github.com/blockchain/blockchain-wallet-v4-frontend/blob/feat/dynamic-erc20/packages/blockchain-wallet-v4-frontend/src/components/Balances/selectors.ts#L340, use token account balances instead
- [x] Include erc20s in top left main balance breakdown
- [x] fee and other components no longer show symbol
- [x] swap/receive icon not appearing
- [x] Remove CoinTypeEnum
- [x] check fees for send eth custodial/non-custodial
- [x] trading account balance is NaN in send dropdown when balance is 0
- [x] fix https://gyazo.com/f56e839242a3da1c34839713dd67dce6
- [x] dynamic download transactions
- [x] Add back originally supported coins to holdings and receive
- [x] Only support the initial coins AAVE, PAX, USDT, WDGLD, YFI
- [x] Empty from/to on tx list for erc20
- [x] Fix send erc20
- [x] Fix tests
- [x] Deprecate txListAppRoute
- [x] leave in tx note stuff but hardcode existing coins
## Phase 2
- [ ] // TODO: erc20 phase 2
- [ ] Remove getTransactionPageHeaderText for erc20s
- [ ] Check that syncing to exchange still works correctly
- [ ] Figure out what to do w/ tx notes for erc20 coins
- [ ] Rename supportedCoins to coinsWithBalance
- [ ] deprecate learn more links
- [ ] dynamic erc20 rates
- [ ] dynamic erc20 transactions page
- [ ] Finish download transactions (relies on historic rates)
- [ ] [defaultCoins](https://github.com/blockchain/blockchain-wallet-v4-frontend/blob/feat/dynamic-erc20/packages/blockchain-wallet-v4-frontend/src/data/prices/sagas.ts#L12-L24)
- [ ] [Check any erc20 balance when sending eth](https://github.com/blockchain/blockchain-wallet-v4-frontend/blob/feat/dynamic-erc20/packages/blockchain-wallet-v4-frontend/src/modals/Eth/SendEth/FirstStep/selectors.js#L36)
## Phase 3
- [ ] Remove all coins from wallet options
- [ ] Deprecate supportedCoins.getOrElse({}) for window.coins coinfigs

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/data/analytics/sagas.ts [hunk 1]
```diff
@@ -60,15 +60,15 @@ export default () => {
...
       )).getOrElse(0)
       const usdtBalance = (yield select(
         selectors.core.data.eth.getErc20Balance,
-        'usdt'
+        'USDT'
       )).getOrElse(0)
       const wdgldBalance = (yield select(
         selectors.core.data.eth.getErc20Balance,
-        'wdgld'
+        'WDGLD'
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/data/analytics/sagas.ts:71]: Remove all this

File: packages/blockchain-wallet-v4-frontend/src/components/Form/SelectBoxEthAddresses/selectors.ts [hunk 2]
```diff
@@ -178,48 +182,16 @@ export const getErc20Data = (
...
-        toUnit: 'YFI'
-      })
-    }
-    return {}
+    const amount = Exchange.convertCoinToCoin({
+      baseToStandard: true,
+      coin,
+      value: data.value
+    })
+    return Exchange.displayPaxToPax({
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/components/Form/SelectBoxEthAddresses/selectors.ts:190]: `return Exchange.displayPaxToPax({` ?

File: packages/blockchain-wallet-v4-frontend/src/data/components/sendBch/sagas.ts [hunk 3]
```diff
@@ -88,10 +88,10 @@ export default ({
         payment = yield payment.from(defaultIndex, ADDRESS_TYPES.ACCOUNT)
         if (to) payment = yield payment.to(to)
         if (amount && amount.coin) {
-          const satAmount = Exchange.convertBchToBch({
+          const satAmount = Exchange.convertCoinToCoin({
             value: amount.coin,
-            fromUnit: 'BCH',
-            toUnit: 'SAT'
+            baseToStandard: false,
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/data/components/sendBch/sagas.ts:93]: can we just default this to false in function so we dont have to pass?

File: packages/blockchain-wallet-v4-frontend/src/modals/Transactions/DownloadTransactions/index.tsx [hunk 4]
```diff
@@ -122,36 +122,17 @@ const mapDispatchToProps = (dispatch: Dispatch, { coin }: OwnProps) => {
     clearTransactions: () =>
       dispatch(actions.core.data[coinCode].clearTransactionHistory()),
     fetchTransactions: (address, startDate, endDate) => {
-      if (coin === 'PAX') {
+      if (coin === 'PAX' || coin === 'USDT' || coin === 'WDGLD') {
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/modals/Transactions/DownloadTransactions/index.tsx:125]: can we do a `coin in Erc20Enum` check?

File: packages/blockchain-wallet-v4-frontend/src/data/components/sendBch/sagas.ts (hunk 3 above)
Comment (schnogz) [reply packages/blockchain-wallet-v4-frontend/src/data/components/sendBch/sagas.ts:93]: or default it to whatever case is used more commingly

Comment (plondon) [reply packages/blockchain-wallet-v4-frontend/src/data/components/sendBch/sagas.ts:93]: sure

File: packages/blockchain-wallet-v4-frontend/src/modals/Transactions/DownloadTransactions/index.tsx (hunk 4 above)
Comment (plondon) [reply packages/blockchain-wallet-v4-frontend/src/modals/Transactions/DownloadTransactions/index.tsx:125]: Sure but neither will be around for long.

Comment (plondon) [reply packages/blockchain-wallet-v4-frontend/src/modals/Transactions/DownloadTransactions/index.tsx]: This file looks at coinfig now

**Pull request merged**